src/panoptoindexconnector/implementations/graph_implementation.py

```diff
@@ -0,0 +1,569 @@
+    Delete Panopto content from the target
+    """
+
+    LOG.info(f"Deleting video from target by id: {video_id}")
+
+    # Get token
+    access_token = get_access_token(config)
+
+    # Set headers
+    headers = {
+        'Authorization': f'Bearer {access_token}'
+    }
+
+    target_address = config.target_address
+    connection_id = config.target_connection["id"]
+    url = f"{target_address}/{connection_id}/items/{video_id}"
+
+    response = requests.delete(url, headers=headers)
+
+    if response.status_code == 200:
+        LOG.info(f"Video ({video_id}) has been deleted from target!")
+    elif response.status_code == 404:
+        LOG.info(f"Video ({video_id}) not found to be delete from target!")
+
+
+#
+# Initialize and teardown steps here
+#
+
+def initialize(config):
+    """
+    Create connection and register schema if not exists
+    """
+
+    try:
+        ensure_connection_availability(config)
+    except Exception as ex:
+        LOG.error(f'Error occurred while initializing graph connector!. Error: {ex}')
+        raise
+
+
+def teardown(config):
+    """
+    Delete token cache file
+    """
+
+    if os.path.exists(TOKEN_CACHE):
+        os.remove(TOKEN_CACHE)
+
+
+#########################################################################
+#
+# Local helpers
+#
+#########################################################################
+
+
+def set_main_fields_to_new_object(video_content):
+    """
+    Set id and content properties to new object
+    """
+
+    return {
+        "id": video_content["Id"],
+        "content": {
+            "type": "text",
+            "value": "{0}{1}".format(video_content["Title"], " " + video_content["Summary"] if video_content["Summary"] else "")
+        }
+    }
+
+
+def set_properties(field_mapping, panopto_content, target_content):
+    """
+    Set properties
+    """
+
+    # Set properties from Info yaml
+    target_content['properties'] = {
+        field:  panopto_content['VideoContent'][key]
+        for key, field in field_mapping['Info'].items()
+        if panopto_content['VideoContent'][key]
+    }
+
+    # Set properties from Metadata yaml
+    target_content["properties"].update({
+        field: panopto_content['VideoContent'][key]
+        for key, field in field_mapping['Metadata'].items()
+        if panopto_content['VideoContent'][key]
+    })
+
+
+def set_principals(config, panopto_content, target_content):
+    """
+    Set principals - Everyone (in tenant), User or Group.
+    If none of permissions are set, target_content will be set to None and skipped for sync
+    """
+
+    if not config.skip_permissions:
+        if is_public_or_all_users_principals(panopto_content):
+            set_principals_to_all(config, target_content)
+        elif is_user_principals(panopto_content):
+            set_principals_to_user(config, panopto_content, target_content)
+    else:
+        set_principals_to_all(config, target_content)
+
+    if not target_content["acl"]:
```

Comment:
You will get a KeyError if `target_content["acl"]` is never set. Try `if "acl" not in target_content or not target_content["acl"]:`

Reply:
We shouldn't get a KeyError for `target_content["acl"]` since we assign empty list in `set_principals_to_user` method and in `set_principals_to_all` list is populated. I changed this part of code with `if not target_content.get("acl"):` to be more clear. Thanks.

Reply:
Isn't there an else condition where the property isn't set at all?

Reply:
To be clear, I  did encounter this when testing against zoom-test.staging yesterday.

Reply:
You are right! There can be else condition where property isn't set. I missed that. My change to use `if not target_content.get("acl")` fixed that case so I think we are safe now. Thanks for clarification!